plugins/power/powerplugin.cpp

```diff
@@ -30,10 +30,43 @@ PowerPlugin::PowerPlugin(QObject *parent)
     : QObject(parent),
 
       m_pluginLoaded(false),
-      m_tipsLabel(new TipsWidget)
+      m_tipsLabel(new TipsWidget),
+      m_uPowerInter(new QDBusInterface("org.freedesktop.UPower",
+                                       "/org/freedesktop/UPower",
+                                       "org.freedesktop.UPower",
+                                       QDBusConnection::systemBus())),
+      m_uBatteryDeviceInter(nullptr)
 {
     m_tipsLabel->setVisible(false);
     m_tipsLabel->setObjectName("power");
+
+    if (!m_uPowerInter->isValid()) {
+        qDebug() << "DBusConnection to org.freedesktop.UPower is invalid";
+        return;
+    }
+
+    QDBusReply<QList<QDBusObjectPath>> reply = m_uPowerInter->call("EnumerateDevices");
+    QList<QDBusObjectPath> paths = reply.value();
+    QDBusObjectPath batteryPath;
+    foreach(auto objectPath, paths) {
+        qDebug() << "EnumerateDevices: ";
+        qDebug() << objectPath.path();
+
+        if (objectPath.path().contains("battery")) {
+            batteryPath = objectPath;
+            break;
+        }
+    }
+
+    m_uBatteryDeviceInter = new QDBusInterface(
```

Comment:
if the path is empty here, you don't have to continue. The rest of the place using the m_uBatteryDeviceInter variable can be checked to skip.

Reply:
Solved in 9958667